liquidity-pool-changes/src/agent.spec.js

```diff
@@ -0,0 +1,171 @@
+const mockToken1Address = '0xFAKETOKEN1ADDRESS'; // .token1()
+const mockPoolAddress = '0xFAKEPOOLADDRESS'; // .address
+const mockToken0Amount = 10000; // token0.balanceOf()
+const mockToken1Amount = 5; // token1.balanceOf()
+const mockToken0Amount2 = 20000; // token0.balanceOf() to produce a large liquidity change
+const mockToken1Amount2 = 10; // token1.balanceOf() to produce a large liquidity change
+const mockPoolBalance = 2;
+const mockDecimals = 3;
+
+const mockPoolContract = {
+  token0: jest.fn().mockResolvedValue(mockToken0Address),
+  token1: jest.fn().mockResolvedValue(mockToken1Address),
+  balanceOf: jest
+    .fn()
+    // failure test cases
+    .mockReturnValueOnce(mockToken0Amount)
+    .mockReturnValueOnce(mockToken1Amount)
+    // success test cases
+    .mockReturnValueOnce(mockToken0Amount)
+    .mockReturnValueOnce(mockToken1Amount)
+    .mockReturnValueOnce(mockToken0Amount2)
+    .mockReturnValueOnce(mockToken1Amount2),
+  decimals: jest.fn().mockResolvedValue(mockDecimals),
+  address: mockPoolAddress,
+};
+
+// combine the mocked provider and contracts into the ethers import mock
+jest.mock('forta-agent', () => ({
+  ...jest.requireActual('forta-agent'),
+  getEthersProvider: jest.fn(),
+  ethers: {
+    ...jest.requireActual('ethers'),
+    providers: {
+      JsonRpcBatchProvider: jest.fn(),
+    },
+    Contract: jest.fn().mockReturnValue(mockPoolContract),
+  },
+}));
+
+const {
+  FindingType,
+  FindingSeverity,
+  Finding,
+} = require('forta-agent');
+
+// axios mocking
+const mockCoinGeckoData = {};
+const mockCoinGeckoResponse = {
+  data: mockCoinGeckoData,
+};
+
+jest.mock('axios', () => ({
+  get: jest.fn().mockResolvedValue(mockCoinGeckoResponse),
+}));
+
+const axios = require('axios');
+
+const {
+  provideHandleBlock,
+  provideInitialize,
+  getPreviousLiquidity,
+} = require('./agent');
+
+// axios mock test
+describe('mock axios GET requests', () => {
+  it('should call axios.get and return the mocked response for CoinGecko', async () => {
+    mockCoinGeckoResponse.data = { '0xtokenaddress': { usd: 1000 } };
+    const response = await axios.get('https://url.url');
+    expect(axios.get).toHaveBeenCalledTimes(1);
+    expect(response.data['0xtokenaddress'].usd).toEqual(1000);
+
+    // reset call count for next test
+    axios.get.mockClear();
+    expect(axios.get).toHaveBeenCalledTimes(0);
+  });
+});
+
+// handler tests
+describe('large liquidity pool change agent', () => {
+  describe('handleBlock', () => {
+    let initializeData;
+    let handleBlock;
+
+    beforeEach(async () => {
+      initializeData = {};
+
+      await provideInitialize(initializeData)();
+
+      handleBlock = provideHandleBlock(initializeData);
+      initializeData.provider = {
+        getBalance: jest.fn().mockResolvedValue(mockPoolBalance),
+      }; // mock provider
+    });
+
+    it('returns empty findings if liquidity change is below given threshold', async () => {
```

Comment:
This test only calls the handler one time, which means it does not calculate any change in liquidity because it only has a one block sample size.  The approach from the next test can be applied here to ensure that a change in liquidity is calculated.